src/Pact/Analyze/Eval/Prop.hs

```diff
@@ -13,7 +14,9 @@ module Pact.Analyze.Eval.Prop where
 
 import           Control.Lens               (Lens', at, ix, view, (%=), (?~))
 import           Control.Monad.Except       (ExceptT, MonadError (throwError))
+#if !MIN_VERSION_base(4,13,0)
 import           Control.Monad.Fail         (MonadFail(..))
+#endif
```

Comment:
[As shown here](https://www.fosskers.ca/en/blog/base#orgafee86), `MonadFail` is exported by `Prelude` in the newest `base`. Prior to that it requires a special import. Since Pact-the-Library needs to support GHCjs as well (which is bound to GHC 8.6 still), we follow [the usual way](https://www.fosskers.ca/en/blog/base#org128075) of supporting this. If we don't like CPP, we can always consider something like `base-compat` as [explained here](https://www.fosskers.ca/en/blog/base#org5f7e98). Or we stop using `MonadFail` so much, but that's a heavier change.

Reply:
To clarify: without the CPP, this import raises a redundant import warning with GHC 8.8. And we have `-Werror` on, so that otherwise fails the build.